flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannel.java

```diff
@@ -270,4 +277,18 @@ public int unsynchronizedGetNumberOfQueuedBuffers() {
 	public String toString() {
 		return "LocalInputChannel [" + partitionId + "]";
 	}
+
+	@Override
+	public EnqueueAction priorityEvent(AbstractEvent event, BufferConsumer eventBufferConsumer) throws IOException {
+		if (inputGate.bufferReceivedListener == null) {
```

Comment:
I think this path should never happen in practice. Although we can not make `bufferReceivedListener` as final in form, we should still guarantee the listener is ready before requesting partition from local channel.

Reply:
Some leftover from POC.